controllers/openstackvmset_controller.go

```diff
@@ -1041,3 +1099,27 @@ func (r *OpenStackVMSetReconciler) getDeletedVMOSPHostnames(instance *ospdirecto
 
 	return annotatedVMs, nil
 }
+
+func (r *OpenStackVMSetReconciler) setProvisioningStatus(instance *ospdirectorv1beta1.OpenStackVMSet, actualState ospdirectorv1beta1.OpenStackVMSetProvisioningStatus) error {
+	// get current ProvisioningStatus
+	currentState := instance.Status.ProvisioningStatus
+
+	// if the current ProvisioningStatus is different from the actual, store the update
+	// otherwise, just log the status again
+	if !reflect.DeepEqual(currentState, actualState) {
+		r.Log.Info(fmt.Sprintf("%s - diff %s", actualState.Reason, diff.ObjectReflectDiff(currentState, actualState)))
+		instance.Status.ProvisioningStatus = actualState
+
+		instance.Status.Conditions = ospdirectorv1beta1.ConditionList{}
+		instance.Status.Conditions.Set(ospdirectorv1beta1.ConditionType(actualState.State), corev1.ConditionTrue, ospdirectorv1beta1.ConditionReason(actualState.Reason), actualState.Reason)
+
+		if err := r.Client.Status().Update(context.TODO(), instance); err != nil {
+			r.Log.Error(err, "Failed to update CR status %v")
+			return err
+		}
+	} else {
+		r.Log.Info(actualState.Reason)
```

Comment:
Same thing here regarding empty string.  For instance, if the reconcile loop keeps hitting the new 663 line, `Reason` is just set to the empty string.  If we just check for this, we can prevent printing an empty `Info` log.

Reply:
I have pushed a commit to not log empty actualState.Reason messages. for vmset and bms